src/zalsa.rs

```diff
@@ -186,21 +188,22 @@ impl Zalsa {
         {
             let jar_type_id = jar.type_id();
             let mut jar_map = self.jar_map.lock();
-            *jar_map
-            .entry(jar_type_id)
-            .or_insert_with(|| {
-                let index = IngredientIndex::from(self.ingredients_vec.len());
-                let ingredients = jar.create_ingredients(self, index);
+            let mut should_create = false;
```

Comment:
I don't quite follow this -- why move this logic out from the `or_insert_with` call?

Reply:
I guess it is so that ingredient construction can call `lookup_jar_by_type`?

Reply:
It'd be nice to have a comment here explaining that interaction

Reply:
I don't remember for sure - as it was weeks before 😅 -, but I think that it was because it had some issues due to creation order like using ingredients index before they are created.
I'll look into such order related things along with the one you commented [here](https://github.com/salsa-rs/salsa/pull/614/files/38ad6555ebe4374bbc69022e005f4136a1839ad9#r1879820347)